From: "K. Y. Srinivasan" <kys@microsoft.com>
Date: Thu, 18 May 2017 10:46:05 -0700
Subject: drivers: hv: vmbus: Increase the time between retries in vmbus_post_msg()
Patch-mainline: submitted (lkml - https://patchwork.kernel.org/patch/9735035/)
References: fate#320485, bnc#1044112

Commit c0bb03924f1a ("Drivers: hv: vmbus: Raise retry/wait limits in
vmbus_post_msg()") increased the retry/wait limits of vmbus_post_msg()
to address the new DoS protection policies in WS2016.
Increase the time between retries to make the
code more robust.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Acked-by: <ohering@suse.de>
---
 drivers/hv/connection.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hv/connection.c b/drivers/hv/connection.c
index 203198bacb27..c9ecc13e9d40 100644
--- a/drivers/hv/connection.c
+++ b/drivers/hv/connection.c
@@ -386,7 +386,7 @@ int vmbus_post_msg(void *buffer, size_t buflen)
 		else
 			mdelay(usec / 1000);
 
-		if (usec < 256000)
+		if (retries < 22)
 			usec *= 2;
 	}
 	return ret;
